### PR TITLE
KVDB function helpers implementation

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#12629|

This PR adds three new function helpers to access KVDB:
- opBuilderKVDBExtract() to perform a mapping action. The Value obtained from the DB is appended to the event.
- opBuilderKVDBMatch() and opBuilderKVDBNotMatch() to perform a filtering action. If the field value is found (or not) in the DB, the event is allowed or blocked.
Also, this PR adds UT for the different usages cases of these Function Helpers

### CODE REVIEW

File: src/engine/source/builder/builders/opBuilderKVDB.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    auto parametersArr {utils::string::split(params, '/')};
+    if (parametersArr.size() != 3) //TODO: We are using default column only now
+    {
+        throw std::runtime_error("Invalid number of parameters for kvdb extracting operator");
+    }
+
+    // Get DB
+    KVDBManager& kvdbManager = KVDBManager::get();
+    KVDB& kvdb = kvdbManager.getDB(parametersArr[1]);
+    if (kvdb.getState() != KVDB::State::Open)
```
Comment (san1526) [inline src/engine/source/builder/builders/opBuilderKVDB.cpp:50]: i think this should rather be a member function of kvdb like `kvdb.isReady()`

File: src/engine/source/builder/builders/opBuilderKVDB.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    {
+        auto msg = fmt::format("[{}] DB isn't available for usage", parametersArr[1]);
+        throw std::runtime_error(std::move(msg));
+    }
+
+    // Get reference key
+    std::string key = std::move(parametersArr[2]);
```
Comment (san1526) [inline src/engine/source/builder/builders/opBuilderKVDB.cpp:57]: just make it a reference or move it in the capture

File: src/engine/source/builder/builders/opBuilderKVDB.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        auto msg = fmt::format("[{}] DB isn't available for usage", parametersArr[1]);
+        throw std::runtime_error(std::move(msg));
+    }
+
+    // Return Lifter
+    return [&kvdb, key, checkExist](types::Observable o)
+    {
+        // Append rxcpp operations
+        return o.filter(
+            [&kvdb, key, checkExist](types::Event e)
```
Comment (san1526) [inline src/engine/source/builder/builders/opBuilderKVDB.cpp:150]: in the future some one could delete the db somewhere else (or update it) so this reference could stay dangling

**Review by san1526 — APPROVED** [review]
minor comments

File: src/engine/source/builder/builders/opBuilderKVDB.cpp (hunk 1 above)
Comment (palaciosjeremias) [reply src/engine/source/builder/builders/opBuilderKVDB.cpp:50]: I like it!
Changed.

File: src/engine/source/builder/builders/opBuilderKVDB.cpp (hunk 2 above)
Comment (palaciosjeremias) [reply src/engine/source/builder/builders/opBuilderKVDB.cpp:57]: Changed to reference.

File: src/engine/source/builder/builders/opBuilderKVDB.cpp (hunk 3 above)
Comment (palaciosjeremias) [reply src/engine/source/builder/builders/opBuilderKVDB.cpp:150]: It's true. We will implement a different approach in https://github.com/wazuh/wazuh/issues/12935 where DB can´t be deleted until every reference is deleted.

**Review by san1526 — APPROVED** [review]
LGTM

**Pull request merged**